packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -591,6 +595,36 @@ abstract class ResidentRunner {
     }
   }
 
+  Future<void> _saveCompilationTrace() async {
+    if (!supportsServiceProtocol)
+      return;
+
+    for (FlutterDevice device in flutterDevices) {
+      for (FlutterView view in device.views) {
+        final int index = device.views.indexOf(view);
+        printStatus('Saving compilation trace for '
+            '${device.device.name}${index == 0 ? '' :'/Isolate$index'}...');
+
+        List<int> buffer;
+        try {
+          buffer = await view.uiIsolate.flutterDumpCompilationTrace();
+          assert(buffer != null);
+        } catch (error) {
+          printError('Error communicating with Flutter on the device: $error');
+          continue;
+        }
+
+        final File outputFile = fs.currentDirectory
+          .childFile('compilation${index == 0 ? '' : index}.txt');
+
+        outputFile.parent.createSync(recursive: true);
+        outputFile.writeAsBytesSync(buffer);
+        printStatus('Compilation trace written to ${fs.path.relative(outputFile.path)}.');
+        return;
```

Comment:
Is this return intentional? Do we want this only for the first view that's successful?

Reply:
Removed for now. It's fine either way since we usually only need one of these files. But it could be useful to have it for other isolates too since there could be differences across FlutterViews.